site/src/pages/components/enhanced-range-input.explainer.mdx

```diff
@@ -0,0 +1,479 @@
+
+```html
+<input type="range" handles="2" min="0" max="100" value="25,75" stepbetween="10">
+```
+4 `list`: Links the range input to a `<datalist>` element, providing tick marks or predefined values.
+
+```html
+<input type="range" min="0" max="100" list="tickmarks">
+<datalist id="tickmarks">
+    <option value="0" label="0%"></option>
+    <option value="25" label="25%"></option>
+    <option value="50" label="50%"></option>
+    <option value="75" label="75%"></option>
+    <option value="100" label="100%"></option>
+</datalist>
+```
+### CSS Properties And Pseudo-elements
+
+To address the current fragmentation and provide a unified styling API, we propose the following pseudo-elements:
+
+1. `::range-track`: Represents the main track of the range input.
+2. `::range-fill`: Represents the filled portion of the track.
+3. `::range-thumb`: Represents the draggable handle(s).
+4. `::range-tick`: Represents individual tick marks on the range input.
+5. `::range-tick-label`: Represents the label associated with each tick mark. This pseudo-element should be able to handle the `content` property for custom content. By default, the content is the value of the tick.
+6. `::range-segment`: Represents sections of the track between handles in multi-handle ranges.
+
+### CSS Selectors To Target Different Pseudo-elements
+
+These pseudo-elements can be used in combination with other CSS selectors, such as `:nth-child()`, to provide granular control over individual ticks, labels, and segments.
+
+The following is an example usage, more details on this are explained below:
+
+```css
+input[type="range"] {
+    appearance: base-range;
+}
+
+input[type="range"]::range-track {
+    height: 4px;
+    background-color: #ddd;
+}
+
+input[type="range"]::range-fill {
+    background-color: #4CAF50;
+}
+
+input[type="range"]::range-thumb {
+    width: 20px;
+    height: 20px;
+    background-color: #2196F3;
+    border-radius: 50%;
+}
+
+/* Style all ticks */
+input[type="range"]::range-tick {
+    width: 2px;
+    height: 8px;
+    background-color: #999;
+}
+
+/* Style every fifth tick differently */
+input[type="range"]::range-tick:nth-child(5n) {
+    height: 12px;
+    background-color: #333;
+}
+/* Style all tick labels */
+input[type="range"]::range-tick-label {
+    font-size: 12px;
+    color: #666;
+}
+
+/* Style labels for every fifth tick differently */
+input[type="range"]::range-tick-label:nth-child(5n) {
+    font-weight: bold;
+    color: #333;
+}
+
+/* Style segments in a multi-handle range */
+input[type="range"][handles="3"]::range-segment:nth-child(1) {
+    background-color: #FF5733;
+}
+
+input[type="range"][handles="3"]::range-segment:nth-child(2) {
+    background-color: #33FF57;
+}
+
+input[type="range"][handles="3"]::range-segment:nth-child(3) {
+    background-color: #3357FF;
+}
+```
+
+### JavaScript API
+
+1. `valueAsNumber`: Returns an array of numbers for multi-handle ranges.
```

Comment:
current spec and implementations expects a single Number so i'm not sure reusing this getter for array of numbers

Reply:
Yeah properties are cheap enough to add I think we should just use a new one rather than overloading the existing one. This property can just return the first value in the index?